packages/graphql/src/schema/create-relationship-fields/create-relationship-fields.ts

```diff
@@ -337,12 +371,14 @@ function createRelationshipFields({
                 },
             });
         }
-        if (rel.settableOptions.onUpdate) {
+        if (rel.settableOptions.onUpdate && nodeFieldUpdateInput) {
             const connectionUpdateInputName = `${rel.connectionPrefix}${upperFieldName}UpdateConnectionInput`;
 
             nodeUpdateInput.addFields({
                 [rel.fieldName]: {
-                    type: rel.typeMeta.array ? `[${nodeFieldUpdateInputName}!]` : nodeFieldUpdateInputName,
+                    type: rel.typeMeta.array
+                        ? `[${nodeFieldUpdateInput.getTypeName()}!]`
+                        : nodeFieldUpdateInput.getTypeName(),
```

Comment:
I think this is an interesting change. Do you think we might be able to use `.getTypeName()` more elsewhere in this file so that we can forgo the need to store the names separately?